### PR TITLE
Add Parser for application and creation of txt file on local device

### PR DESCRIPTION
**Pull request by itszhixuan**
Parser functions that reads in user inputs to implement features based on specific commands available to user. 

Parsing includes:
1. Main parser that reads in all inputs
2. Parser support for add
3. Parser support for delete
4. Parser support for editing
5. Parser support for loans and borrows

### CODE REVIEW

File: src/main/java/seedu/Parser/ParseEdit.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class ParseEdit {
+    public static Command editItem(String line) {
+        // Format: edit, index, d/date, a/amount, s/description
+        int space = line.indexOf(" ");
+        String[] allDetails = SplitCommand.splitCommand(space, line);
+        String details = allDetails[1];
+
+        int posDSlash = details.indexOf('/');
+        String[] furtherDetails = SplitCommand.splitCommand(posDSlash, details);
+        String indexVal = furtherDetails[0];
```
Comment (dsicol) [inline src/main/java/seedu/Parser/ParseEdit.java:16]: Moving forward can use constants to denote this number to avoid magic literals

**Comment by dsicol** [comment]
LGTM

File: src/main/java/seedu/Parser/ParseEdit.java (hunk 1 above)
Comment (itszhixuan) [reply src/main/java/seedu/Parser/ParseEdit.java:16]: noted, will change it in my next update

**Review by dsicol — APPROVED** [review]
LGTM

**Pull request merged**